src/lib/shaders/hit-test-shaders.ts

```diff
@@ -27,25 +27,48 @@ import * as ShaderFunctions from './shader-functions';
 
 /**
  * Generate the fragment (pixel) shader for the hit test command. For each
- * sprite, this shader writes whether the screen pixel of interest intersects it
- * to the RGB color channels of the output texel.
+ * sprite, this shader writes a packed float value in the range 0-1 into the
+ * output Uint8 RGBA channels. To unpack, multiply by capacity+1, then subtract
+ * 1. This will yield a number in the range (-1,capacity - 1), which is either
+ * -1 for a miss, or the index of the SpriteImpl.
+ *
+ * @see http://marcodiiga.github.io/encoding-normalized-floats-to-rgba8-vectors
  */
 export function fragmentShader() {
   return glsl`
 precision lowp float;
 
+// Need to know the maximum possible value for the index of the Sprite to
+// normalize the value for RGBA packing.
+uniform float capacity;
+
 varying float varyingHitTestResult;
 
+vec4 packFloatToVec4i(const float value) {
```

Comment:
nit: float style is inconsistent throughout this file. The dominant style appears to be `.5` or `0.`, but this function uses `0.0`.

Reply:
Fixing.